### DISCUSSION TITLE
YOLOv8 keypoints like mediapipe

### DISCUSSION THREAD

**Discussion by corones777** [ultralytics/ultralytics#4316] [Docs]
YOLO!?

I am working with YOLOv8 and need to access the pose estimation keypoints x and y positions for each point on the persons being tracked.  For example I want to write code that places a rectangle at the nose position landmark as a center point vertex.  I have used Mediapipe and had great success in writing code that draws rectangles at nose, hands and foot keypoints.  Mediapipe only allows tracking of one person at a time, however it has all the functionality I need to perform computational geometry for the computer vision basketball program demonstrated in the youtube link below.  I have tried everything and am stumped…

Python

https://www.youtube.com/watch?v=-yvZ-_nKDbY 

Thank you!

**Comment by glenn-jocher** [comment]
@corones777 hello!

Great to hear about your interest in pose estimation with YOLOv8! YOLOv8 indeed supports pose/keypoint estimation tasks, which can be used to identify and track various keypoints on a person.

To achieve the functionality similar to what you've experienced with Mediapipe, you would need to train a YOLOv8 model on a dataset that includes annotations for the keypoints you're interested in, such as nose, hands, and feet. Once the model is trained, you can use the Predict mode to detect these keypoints in your images or video streams.

For placing a rectangle at the nose position, you would extract the x and y coordinates of the nose keypoint from the model's predictions and then use your preferred method to draw the rectangle in your application.

If you're looking for more detailed instructions on how to train your model for pose estimation, validate it, and make predictions, please refer to our documentation on pose/keypoint estimation tasks. The documentation will guide you through the necessary steps to get your model up and running for your specific use case.

Keep in mind that YOLOv8 is designed to track multiple objects (and by extension, multiple people) simultaneously, which should suit your requirement for tracking more than one person at a time.

If you encounter any specific issues or have further questions, feel free to reach out. The community and the Ultralytics team are here to help. Good luck with your basketball program! 🏀🤖
